src/main/java/org/ensime/maven/plugins/ensime/ServerIndexMojo.java

```diff
@@ -0,0 +1,145 @@
+package org.ensime.maven.plugins.ensime;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.ensime.maven.plugins.ensime.model.EnsimeConfig;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+/**
+ * Uses maven project metadata to run the ensime server in index-only mode to pre-index the
+ * codebase.
+ *
+ * @author Aaditya Ramesh
+ */
+@Mojo(
+    name = "serverIndex",
```

Comment:
would it be better to name it `ensimeServerIndex`?

Reply:
Yes, I was initially thinking it may be better to call it ensimeServerIndex, but since we're going to be invoking the ensime plugin using `mvn ensime:...` anyway, it might be less redundant to call it `serverIndex`. The reason sbt uses the full `ensimeServerIndex` is because ensimeServerIndex must be globally unique, whereas with the maven ensime plugin, we already know that we're attempting to invoke a goal from the ensime plugin so I thought it's more concise to just use serverIndex. What do you think? If we expand it, it would be invoked like `mvn ensime:ensimeServerIndex` which seems a bit strange to me.

Reply:
That's fine then. Frankly did not know you have to prefix plugin name in maven.